test/e2e/apps/deployment.go

```diff
@@ -416,7 +413,9 @@ var _ = SIGDescribe("Deployment", func() {
 			case watch.Modified:
 				if deployment, ok := event.Object.(*appsv1.Deployment); ok {
 					found := deployment.ObjectMeta.Name == testDeployment.Name &&
-						deployment.ObjectMeta.Labels["test-deployment-static"] == "true"
```

Comment:
We don't need to remove this, `deployment.ObjectMeta.Labels["test-deployment-static"]` shoud be `true`